src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2713,30 +2713,30 @@ private bool TryVisitConditionalAccess(BoundExpression node, [NotNullWhen(true)]
 
             stateWhenNotNull = default;
             return false;
-
         }
 
         /// <summary>
         /// "State when not null" can only propagate out of a conditional access if
         /// it is not subject to a user-defined conversion whose parameter is not of a non-nullable value type.
         /// </summary>
-        protected static bool CanPropagateStateWhenNotNull(BoundConditionalAccess operand, Conversion conversion)
+        protected static bool CanPropagateStateWhenNotNull(Conversion conversion)
         {
-            if (conversion.Kind is not (ConversionKind.ImplicitUserDefined or ConversionKind.ExplicitUserDefined))
+            if (!conversion.IsUserDefined)
             {
                 return true;
             }
 
             var method = conversion.Method;
-            Debug.Assert(method is not null);
-            Debug.Assert(method.ParameterCount is 1);
-            var param = method.Parameters[0];
-            if (operand.Type.IsNullableType() && param.Type.IsNonNullableValueType())
+            if (method is null)
             {
-                return true;
+                // This can happen when the user-defined conversion to use is ambiguous.
+                // For example in `SyntaxBinderTests.TestNullCoalesceWithInvalidUserDefinedConversions_03`
+                return false;
             }
 
-            return false;
+            Debug.Assert(method.ParameterCount is 1);
```

Comment:
Is this actually true for all error scenarios?

Reply:
And if it isn't, what would we even do about it?

Reply:
> Is this actually true for all error scenarios?

I actually don't know. I'll add a test.

> And if it isn't, what would we even do about it?

I think we're in "degrade gracefully" mode at that point. I would be fine with just using the first parameter if one exists, and otherwise saying that the conversion can't propagate "state when not null".

Reply:
Fixing this method by imitating logic from VisitUserDefinedConversion.